### PR TITLE
First draft of OpenStack-Ceph config

### PR DESCRIPTION
**Pull request by oneswig**

### CODE REVIEW

**Review by jovial — COMMENTED** [review]
shoudn't this be `/etc/ceph/ceph.conf` ?

File: etc/kayobe/kolla/config/cinder/cinder-volume/ceph.client.cinder.keyring [hunk 1]
```diff
@@ -0,0 +1,2 @@
+client.cinder
```
Comment (jovial) [inline etc/kayobe/kolla/config/cinder/cinder-volume/ceph.client.cinder.keyring]: Shouldn't this be in square brackets?

File: etc/kayobe/kolla/config/glance/glance-api.conf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[glance_store]
+stores = rbd
+default_store = rbd
+rbd_store_pool = images
+rbd_store_user = glance
+rbd_store_ceph_conf = /etc/ceph/sib-dev.conf
```
Comment (jovial) [inline etc/kayobe/kolla/config/glance/glance-api.conf]: Shouldn't this be `etc/ceph/ceph.conf`?

File: etc/kayobe/kolla/config/glance/ceph.client.glance.keyring [hunk 3]
```diff
@@ -0,0 +1,2 @@
+client.glance
```
Comment (jovial) [inline etc/kayobe/kolla/config/glance/ceph.client.glance.keyring]: square brackets?

**Comment by oneswig** [comment]
Good spots thanks Will - I had wanted a non-default name but according to Kolla documentation the Ceph config appears to be fixed to the default name of ceph.

**Review by jovial — APPROVED** [review]

File: etc/kayobe/kolla/config/cinder/cinder-volume.conf [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+enabled_backends=ceph-rbd
+
+[ceph-rbd]
+rbd_ceph_conf=/etc/ceph/ceph.conf
+rbd_user=cinder
+backend_host=rbd:volumes
+rbd_pool=volumes
+volume_backend_name=ceph-rbd
+volume_driver=cinder.volume.drivers.rbd.RBDDriver
+rbd_secret_uuid = {{ cinder_rbd_secret_uuid }}
```
Comment (markgoddard) [inline etc/kayobe/kolla/config/cinder/cinder-volume.conf:11]: Is this in kayobe config, or does it need to be marked `{% raw %}`?

File: etc/kayobe/kolla/config/cinder/ceph.conf [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[global]
+fsid = 822209f7-5dfb-427c-b3d0-582e8549a218
+mon host = 10.12.5.5,10.12.5.6,10.12.5.7
+mon initial members = ceph-0,ceph-1,ceph-2
+auth_cluster_required = cephx
+auth_service_required = cephx
+auth_client_required = cephx
```
Comment (markgoddard) [inline etc/kayobe/kolla/config/cinder/ceph.conf:7]: At GR we found that some of the nova config was also required, as it gets used when mounting the volume.

Comment (markgoddard) [reply etc/kayobe/kolla/config/cinder/ceph.conf:7]: I think we kept `nova_backend_ceph = no`, and did not use ceph.client.nova.keyring but had ceph.client.cinder.keyring in etc/kayobe/kolla/config/nova

Comment (markgoddard) [reply etc/kayobe/kolla/config/cinder/ceph.conf:7]: There may have been changes in stackhpc/queens for kolla-ansible also to make this work, not sure if they made it to stackhpc/pike.

Comment (markgoddard) [reply etc/kayobe/kolla/config/cinder/ceph.conf:7]: It's 355ce7e539feb02ef51d8e6ea5b89fc7541948d6 on kolla-ansible stackhpc/queens, looks like it needs a port to stackhpc/pike.

**Pull request merged**